components/scream/src/physics/shoc/atmosphere_macrophysics.cpp

```diff
@@ -287,7 +288,7 @@ void SHOCMacrophysics::initialize_impl ()
   const Real z_surf = 0.0;
 
   shoc_preprocess.set_variables(m_num_cols,m_num_levs,m_num_tracers,z_surf,m_cell_area,
-                                T_mid,p_mid,pseudo_density,omega,phis,surf_sens_flux,surf_latent_flux,
+                                T_mid,p_mid,p_int,pseudo_density,omega,phis,surf_sens_flux,surf_latent_flux,
                                 surf_mom_flux,qv,qc,tke,tke_copy,z_mid,z_int,cell_length,
                                 dse,rrho,rrho_i,thv,dz,zt_grid,zi_grid,wpthlp_sfc,wprtp_sfc,upwp_sfc,vpwp_sfc,
                                 wtracer_sfc,wm_zt,inv_exner,thlm,qw);
```

Comment:
A few lines down there is 
```
input.presi  = get_field_in("p_int").get_view<const Spack**>();
```

Could you change this to 
```
input.presi  = p_int;
```
since we are now declaring `p_int`?